src/components/manager.ts

```diff
@@ -358,6 +358,8 @@ export class Manager {
                 await this.parseFlsFile(this.rootFile)
                 this.extension.structureProvider.refresh()
                 this.extension.structureProvider.update()
+                this.extension.builder.reset()
```

Comment:
I don't like resetting `previouslyUsedRecipe` every time rootFile changes. It make `"latex-workshop.latex.recipe.default": "lastUsed"` useless.

Reply:
Right, I have remove my last commit. I am not a big fan of this `lastUsed` recipe feature. 

https://github.com/James-Yu/LaTeX-Workshop/blob/0ce09bd80e65ffd41658b7b96c917fc007c3feb5/src/components/builder.ts#L429-L446

When `"latex-workshop.latex.recipe.default": "lastUsed"` and the last used recipe name is no more in the recipe list, we peek the first recipe. I think it is a good solution and at least it will not change the current behaviour for single root workspace, which is very important.